app/objects/c_operation.py

```diff
@@ -242,35 +247,30 @@ def report(self, file_svc, output=False, redacted=False):
 
     """ PRIVATE """
 
-    async def _run_phases(self, planner):
-        for phase in self.adversary.phases:
+    async def _run(self, planner):
+        handle = self.adversary.atomic_ordering
+        if not self.atomic_enabled:
+            handle = [handle]
+        for ability in handle:
             if not await self.is_closeable():
-                await planner.execute(phase)
+                await planner.execute(handle.index(ability))
                 if planner.stopping_condition_met:
```

Comment:
This seems a little hacky. From what I'm reading, if atomic isn't enabled, the adversary's ability list is a list of a list?

Reply:
Commented this to explain things a little bit.